edge-util/src/Microsoft.Azure.Devices.Edge.Util/Option.cs

```diff
@@ -202,6 +202,25 @@ public Option<T> Filter(Func<T, bool> predicate)
 
     public static class Option
     {
+        public static IEnumerable<T> FilterMap<T>(this IEnumerable<Option<T>> source, Func<T, bool> predicate)
+        {
+            Preconditions.CheckNotNull(source, nameof(source));
+            Preconditions.CheckNotNull(predicate, nameof(predicate));
+
+            foreach (var item in source)
+                if (item.Filter(predicate).HasValue)
+                    yield return item.OrDefault();
+        }
+
+        public static IEnumerable<T> FilterMap<T>(this IEnumerable<Option<T>> source)
+        {
+            Preconditions.CheckNotNull(source, nameof(source));
+
+            foreach (var item in source)
+                if (item.HasValue)
+                    yield return item.OrDefault();
```

Comment:
```suggestion
                    yield return item.Value;
```

Reply:
'Value' property is not accessible by the extension method, because it is private. The two options are OrDefault() or Expect(), where Expect() needs a function creating an exception when there is no value. As I check and ensure that always there is a value (previous 'if'), I figured that using OrDefault() is safe and seemed more simple.